SharedSource/Core/Utilities/ReferenceEqualityComparer.cs

```diff
@@ -28,15 +29,15 @@ namespace Remotion.Utilities
   /// and not just equal for reference checking purposes.      
   /// </summary>      
   /// <typeparam name="T">the type of object to check</typeparam>  
-  partial class ReferenceEqualityComparer<T> : IEqualityComparer<T>
+  partial class ReferenceEqualityComparer<T> : IEqualityComparer<T> where T: notnull

```

Comment:
My bad. I think a `class` constraint and `public bool Equals (T? x, T? y)` would be better. what do you think?